certbot/tests/util_test.py

```diff
@@ -473,74 +474,98 @@ def test_wildcard(self):
+
+    @mock.patch("warnings.warn")
+    @mock.patch("certbot.util.distro")
+    @unittest.skipUnless(sys.platform.startswith("linux"), "requires Linux")
+    def test_get_systemd_os_info_deprecation(self, _, mock_warn):
+        import certbot.util as cbutil
+        cbutil._USE_DISTRO = True  # pylint: disable=protected-access
+        cbutil.get_systemd_os_info()
+        self.assertTrue(mock_warn.called)
 
     @mock.patch("certbot.util.subprocess.Popen")
     def test_non_systemd_os_info(self, popen_mock):
-        from certbot.util import (get_os_info, get_python_os_info,
-                                     get_os_info_ua)
-        with mock.patch('certbot.compat.os.path.isfile', return_value=False):
-            with mock.patch('platform.system_alias',
-                            return_value=('NonSystemD', '42', '42')):
-                self.assertEqual(get_os_info()[0], 'nonsystemd')
-                self.assertEqual(get_os_info_ua(),
-                                 " ".join(get_python_os_info()))
-
-            with mock.patch('platform.system_alias',
-                            return_value=('darwin', '', '')):
-                comm_mock = mock.Mock()
-                comm_attrs = {'communicate.return_value':
-                              ('42.42.42', 'error')}
-                comm_mock.configure_mock(**comm_attrs)
-                popen_mock.return_value = comm_mock
-                self.assertEqual(get_os_info()[0], 'darwin')
-                self.assertEqual(get_os_info()[1], '42.42.42')
-
-            with mock.patch('platform.system_alias',
-                            return_value=('linux', '', '')):
-                with mock.patch('platform.linux_distribution',
-                                side_effect=AttributeError,
-                                create=True):
-                    with mock.patch('distro.linux_distribution',
-                                    return_value=('', '', '')):
-                        self.assertEqual(get_python_os_info(), ("linux", ""))
-
-                    with mock.patch('distro.linux_distribution',
-                                    return_value=('testdist', '42', '')):
-                        self.assertEqual(get_python_os_info(), ("testdist", "42"))
-
-            with mock.patch('platform.system_alias',
-                            return_value=('freebsd', '9.3-RC3-p1', '')):
-                self.assertEqual(get_python_os_info(), ("freebsd", "9"))
-
-            with mock.patch('platform.system_alias',
-                            return_value=('windows', '', '')):
-                with mock.patch('platform.win32_ver',
-                                return_value=('4242', '95', '2', '')):
-                    self.assertEqual(get_python_os_info(),
-                                     ("windows", "95"))
+        import certbot.util as cbutil
+        cbutil._USE_DISTRO = False  # pylint: disable=protected-access
```

Comment:
I'd argue that the problems implementing my suggestion at https://github.com/certbot/certbot/pull/7467/files/a1c1c3d36a7c343697e549e60b0bfc4258371c71..6654f3b9c7b644ea520d69c3adc33ae754b9e2f7#r340904031 are due to this test not properly cleaning up after itself. To fix this, I think we can change this line to:
```suggestion
        with mock.patch('certbot.util._USE_DISTRO', False):
```
and indent the rest of this function which should allow you to make the changes in https://github.com/certbot/certbot/pull/7467/files/a1c1c3d36a7c343697e549e60b0bfc4258371c71..6654f3b9c7b644ea520d69c3adc33ae754b9e2f7#r340904031.

Reply:
This is definitely the non-sloppy way of handling the issue, thanks!